dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task04305UpdateWorkflowActionTable.java

```diff
@@ -0,0 +1,596 @@
+        if (DbConnectionFactory.isPostgres()) {
+            return POSTGRES_CREATE_WORKFLOW_SCHEME_X_STRUCTURE_INDEX;
+        } else if (DbConnectionFactory.isOracle()) {
+            return ORACLE_CREATE_WORKFLOW_SCHEME_X_STRUCTURE_INDEX;
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Drops the {@code workflow_scheme_x_structure} unique index.
+     * @return
+     */
+    private String dropTableIndex() {
+        if (DbConnectionFactory.isMySql()) {
+            return MYSQL_DROP_WORKFLOW_SCHEME_X_STRUCTURE_INDEX;
+        } else if (DbConnectionFactory.isPostgres()) {
+            return POSTGRES_DROP_WORKFLOW_SCHEME_X_STRUCTURE_INDEX;
+        } else if (DbConnectionFactory.isMsSql()) {
+            return MSSQL_DROP_WORKFLOW_SCHEME_X_STRUCTURE_INDEX;
+        } else if (DbConnectionFactory.isOracle()) {
+            return ORACLE_DROP_WORKFLOW_SCHEME_X_STRUCTURE_INDEX;
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Verifies if the {@code scheme_id} column already exists in the {@code workflow_action} table.
+     * @return
+     */
+    private String findSchemeIdColumn() {
+        if (DbConnectionFactory.isMySql()) {
+            return MYSQL_FIND_SCHEME_ID_COLUMN;
+        } else if (DbConnectionFactory.isPostgres()) {
+            return POSTGRES_FIND_SCHEME_ID_COLUMN;
+        } else if (DbConnectionFactory.isMsSql()) {
+            return MSSQL_FIND_SCHEME_ID_COLUMN;
+        } else if (DbConnectionFactory.isOracle()) {
+            return ORACLE_FIND_SCHEME_ID_COLUMN;
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Verifies if the {@code requires_checkout_option} column already exists in the {@code workflow_action} table.
+     * @return
+     */
+    private String findRequiresCheckoutOptionsColumn() {
+        if (DbConnectionFactory.isMySql()) {
+            return MYSQL_FIND_REQUIRES_CHECKOUT_OPTION_COLUMN;
+        } else if (DbConnectionFactory.isPostgres()) {
+            return POSTGRES_FIND_REQUIRES_CHECKOUT_OPTION_COLUMN;
+        } else if (DbConnectionFactory.isMsSql()) {
+            return MSSQL_FIND_REQUIRES_CHECKOUT_OPTION_COLUMN;
+        } else if (DbConnectionFactory.isOracle()) {
+            return ORACLE_FIND_REQUIRES_CHECKOUT_OPTION_COLUMN;
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Create the PK constraint for the intermediate table.
+     * @return
+     */
+    private String createIntermediateTablePk() {
+        if (DbConnectionFactory.isMySql()) {
+            return MYSQL_CREATE_INTERMEDIATE_TABLE_PK;
+        } else if (DbConnectionFactory.isPostgres()) {
+            return POSTGRES_CREATE_INTERMEDIATE_TABLE_PK;
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Verifies if the intermediate {@code workflow_action_step} table already exists.
+     * @return
+     */
+    private String findIntermediateTable() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this method 5, down to the maximum allowed 3. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1142)